packages/supi/src/install/index.ts

```diff
@@ -401,6 +403,41 @@ export async function mutateModules (
       })
     }
 
+    if (maybeOpts.strict) {
+      const project = projectsToInstall[0]
+      let dependenciesList: string[] = []
+      switch (project.mutation) {
+        case 'installSome': {
+          if (project.allowNew === false) {
+            for (const p of projectsToInstall) {
+              dependenciesList = [...p.wantedDependencies.map(wantedDependency => wantedDependency.raw), ...dependenciesList]
```

Comment:
you should select `wantedDependency.alias`, not `wantedDependency.raw`. `wantedDependency.raw` may be a name and version spec, like `express@next`.